File: mbtsdk/src/main/java/eventbus/events/FirmwareUpdateClientEvent.java
package eventbus.events;

import core.device.oad.OADState;
import engine.clientevents.BaseError;

/**
 * Event posted when the OAD firmware update state changes or when an error occurs during the update.
 *
 * @author Sophie Zecri on 24/07/2019
 */
public class FirmwareUpdateClientEvent {

    private final static int UNDEFINED = -1;

    private BaseError error;
    private String additionalInfo;
    private OADState oadState;
    private int oadProgress = UNDEFINED;

    public FirmwareUpdateClientEvent(BaseError error, String additionalInfo) {
        this.error = error;
        this.additionalInfo = additionalInfo;
    }

    public FirmwareUpdateClientEvent(OADState oadState) {
        this.oadState = oadState;
        this.oadProgress = oadState.convertToProgress();
    }

    public FirmwareUpdateClientEvent(int oadProgress) {
        this.oadProgress = oadProgress;
    }

    public BaseError getError() {
        return error;
    }

    public String getAdditionalInfo() {
        return additionalInfo;
    }

    public OADState getOadState() {
        return oadState;
    }

    public int getOadProgress() {
        return oadProgress;
    }

    @Override
    public String toString() {
        return "FirmwareUpdateClientEvent{" +
                "error=" + error +
                ", additionalInfo='" + additionalInfo + '\'' +
                ", oadState=" + oadState +
                ", oadProgress=" + oadProgress +
                '}';
    }
}

File: mbtsdk/src/main/java/eventbus/events/FirmwareUpdateClientEvent.kt
package eventbus.events

import core.device.oad.OADState
import engine.clientevents.BaseError

/**
 * Event posted when the OAD firmware update state changes or when an error occurs during the update.
 *
 * @author Sophie Zecri on 24/07/2019
 */
class FirmwareUpdateClientEvent : IEvent {
  var error: BaseError? = null
    private set
  var additionalInfo: String? = null
    private set
  var oadState: OADState? = null
    private set
  var oadProgress = UNDEFINED
    private set

  constructor(error: BaseError?, additionalInfo: String?) {
    this.error = error
    this.additionalInfo = additionalInfo
  }

  constructor(oadState: OADState) {
    this.oadState = oadState
    oadProgress = oadState.convertToProgress()
  }

  constructor(oadProgress: Int) {
    this.oadProgress = oadProgress
  }

  override fun toString(): String {
    return "FirmwareUpdateClientEvent{" +
        "error=" + error +
        ", additionalInfo='" + additionalInfo + '\'' +
        ", oadState=" + oadState +
        ", oadProgress=" + oadProgress +
        '}'
  }

  companion object {
    private const val UNDEFINED = -1
  }
}